services/leecher/leecher/leecher.py

```diff
@@ -24,18 +23,51 @@ def create_event_description(event):
 
 
 def callback(event):
-    if event.topic in IGNORE_TOPICS:
+    """Callback function for Aquarium event listener.
+    Checks the event data for item, gets corresponding project, then dispatches an event to Ayon.
+    """
+
+    # get aq item from event
+    event_data = event.to_dict()
+    item = event_data["data"].get("item")
+    item_key = item.get("_id")
+    if not all([item, item_key]):
+        log.error(f"Can't retrieve Aquarium item from event: {event_data}")
         return
+    aq_item = _AQS.aq.cast(item)
 
-    ayon_api.dispatch_event(
+    # get project from item
+    query = "# <($Child, 5)- item.type == 'Project'"
+    projects: List[dict] = aq_item.traverse(meshql=query)
+    if len(projects) > 1:
+        log.error(f"More than one project found for item: {item_key}")
+        return
+    if len(projects) == 0:
+        log.error(f"No project found for item: {item_key}")
+        return
+    aq_project = _AQS.aq.cast(projects[0].get("item"))
```

Comment:
```suggestion
    aq_project = context.get('project')
    
    if not aq_project:
        log.error(f"Can't retrieve Aquarium project from event: #{event._key}")
        return
```

Reply:
addressed in https://github.com/ynput/ayon-aquarium/commit/f8301de725faf409bb99514434dc91a2de448dfa